### PR TITLE
implement low hanging clippy suggestions for the common crate

### PR DESCRIPTION
**Pull request by mpeck**
Implements many of the low hanging clippy suggestions for the common crate. There are a few other warning that clippy outputs, but I think those might be warrant their own PR.

### CODE REVIEW

**Comment by thesentinels** [comment]
Thanks for the pull request! Here is what will happen next:
 1. Your PR will be reviewed by the maintainers
 2. If everything looks good, one of them will `approve` it, and your PR will be merged.

Thank you for contributing!

File: components/common/src/command/package/install.rs [hunk 1]
```diff
@@ -285,10 +282,12 @@ impl<'a> FullyQualifiedPackageIdent<'a> {
...
-        self.ident.as_ref().archive_name().expect(&format!(
-            "PackageIdent {} should be fully qualified",
-            self.ident.as_ref()
-        ))
+        self.ident.as_ref().archive_name().unwrap_or_else(|_| {
+            panic!(
+                "PackageIdent {} should be fully qualified",
+                self.ident.as_ref()
+            )
+        })
```
Comment (mpeck) [inline components/common/src/command/package/install.rs]: https://rust-lang.github.io/rust-clippy/master/#expect_fun_call

File: components/common/src/command/package/install.rs [hunk 2]
```diff
@@ -124,8 +124,8 @@ impl FromStr for InstallSource {
             let target = archive.target()?;
             match archive.ident() {
                 Ok(ident) => Ok(InstallSource::Archive(LocalArchive {
-                    ident: ident,
-                    target: target,
+                    ident,
+                    target,
```
Comment (mpeck) [inline components/common/src/command/package/install.rs]: https://rust-lang.github.io/rust-clippy/master/#redundant_field_names

File: components/common/src/command/package/install.rs [hunk 3]
```diff
@@ -149,10 +149,7 @@ impl FromStr for InstallSource {
                 // one string, such as `"x86_64-linux::core/redis"` (or similar). As there is
                 // currently no such representation, I'd argue that this `FromStr` is no longer
                 // reasonable. However, it's doing the job for now and we can proceed with caution.
-                Ok(ident) => Ok(InstallSource::Ident(
-                    ident,
-                    PackageTarget::active_target().clone(),
-                )),
+                Ok(ident) => Ok(InstallSource::Ident(ident, *PackageTarget::active_target())),
```
Comment (mpeck) [inline components/common/src/command/package/install.rs]: https://rust-lang.github.io/rust-clippy/master/#clone_on_copy

File: components/common/src/command/package/install.rs [hunk 4]
```diff
@@ -420,11 +421,11 @@ impl<'a> InstallTask<'a> {
     /// fully-qualified identifier of package that was infstalled
     /// (which, as we have seen, may not be the same as the identifier
     /// that was passed in).
-    fn from_ident<T>(
+    fn install_from_ident<T>(
         &self,
         ui: &mut T,
         ident: PackageIdent,
-        target: &PackageTarget,
+        target: PackageTarget,
```
Comment (mpeck) [inline components/common/src/command/package/install.rs]: https://rust-lang.github.io/rust-clippy/master/#trivially_copy_pass_by_ref

File: components/common/src/command/package/install.rs [hunk 5]
```diff
@@ -452,7 +453,7 @@ impl<'a> InstallTask<'a> {
 
     /// Given an archive on disk, ensure that it is properly installed
     /// and return the package's identifier.
-    fn from_archive<T>(
+    fn install_from_archive<T>(
```
Comment (mpeck) [inline components/common/src/command/package/install.rs]: https://rust-lang.github.io/rust-clippy/master/#wrong_self_convention

File: components/common/src/command/package/install.rs [hunk 6]
```diff
@@ -695,19 +694,18 @@ impl<'a> InstallTask<'a> {
...
-                || self.fetch_artifact(ui, ident, target, token),
-                |res| res.is_ok(),
-            ).is_err()
-            {
-                return Err(Error::DownloadFailed(format!(
-                    "We tried {} times but could not download {}. Giving up.",
-                    RETRIES, ident
-                )));
-            }
+        } else if retry(
```
Comment (mpeck) [inline components/common/src/command/package/install.rs]: https://rust-lang.github.io/rust-clippy/master/#collapsible_if

File: components/common/src/command/package/install.rs [hunk 7]
```diff
@@ -809,18 +807,16 @@ impl<'a> InstallTask<'a> {
         {
             let mut artifact = PackageArchive::new(&file);
             let artifact_ident = artifact.ident().ok();
-            if let None = artifact_ident {
+            if artifact_ident.is_none() {
```
Comment (mpeck) [inline components/common/src/command/package/install.rs]: https://rust-lang.github.io/rust-clippy/master/#redundant_pattern_matching

File: components/common/src/command/package/install.rs [hunk 8]
```diff
@@ -931,13 +927,13 @@ impl<'a> InstallTask<'a> {
         // Canonicalize both paths to ensure that there aren't any symlinks when comparing them
         // later. These calls can fail under certain circumstances, so we warn, allow failure and
         // try to continue.
-        let artifact_path = if let Some(canonicalized) = artifact_path.canonicalize().ok() {
+        let artifact_path = if let Ok(canonicalized) = artifact_path.canonicalize() {
```
Comment (mpeck) [inline components/common/src/command/package/install.rs]: https://rust-lang.github.io/rust-clippy/master/#if_let_some_result

File: components/common/src/command/package/install.rs [hunk 9]
```diff
@@ -991,13 +987,13 @@ impl<'a> InstallTask<'a> {
         let active_target = PackageTarget::active_target();
         if active_target != &artifact_target {
             return Err(Error::HabitatCore(hcore::Error::WrongActivePackageTarget(
-                active_target.clone(),
-                artifact_target.clone(),
+                *active_target,
```
Comment (mpeck) [inline components/common/src/command/package/install.rs]: https://rust-lang.github.io/rust-clippy/master/#clone_on_copy

File: components/common/src/command/package/install.rs [hunk 10]
```diff
@@ -1075,10 +1071,11 @@ impl<'a> InstallTask<'a> {
...
         for channel in channels.iter().map(|c| Channel::new(c)) {
-            match self.fetch_latest_pkg_ident_in_channel_for(ident, target, &channel, token) {
-                Ok(pkg) => res.push((channel.to_string(), format!("{}", pkg))),
-                Err(_) => (),
-            };
+            if let Ok(pkg) =
+                self.fetch_latest_pkg_ident_in_channel_for(ident, target, &channel, token)
+            {
+                res.push((channel.to_string(), format!("{}", pkg)))
+            }
```
Comment (mpeck) [inline components/common/src/command/package/install.rs]: https://rust-lang.github.io/rust-clippy/master/#single_match

File: components/common/src/defaults.rs [hunk 11]
```diff
@@ -17,20 +17,20 @@
 //! Eventually this will be composed of fully typed default values. But as a first step we
 //! need a spot to consolidate those values and help simplify some of the logic around them.
 
-pub const GOSSIP_DEFAULT_IP: &'static str = "0.0.0.0";
+pub const GOSSIP_DEFAULT_IP: &str = "0.0.0.0";
```
Comment (mpeck) [inline components/common/src/defaults.rs]: https://rust-lang.github.io/rust-clippy/master/#const_static_lifetime

File: components/common/src/error.rs [hunk 12]
```diff
@@ -62,17 +62,17 @@ impl fmt::Display for Error {
                 ai, a, i
             ),
             Error::CantUploadGossipToml => {
-                format!("Can't upload gossip.toml, it's a reserved file name")
+                "Can't upload gossip.toml, it's a reserved file name".to_string()
```
Comment (mpeck) [inline components/common/src/error.rs]: https://rust-lang.github.io/rust-clippy/master/#useless_format

File: components/common/src/package_graph.rs [hunk 13]
```diff
@@ -83,7 +83,7 @@ impl PackageGraph {
     }
 
     /// Extend a graph by adding in dependencies for a package
-    fn extend(&mut self, package: &PackageIdent, deps: &Vec<PackageIdent>) -> (usize, usize) {
+    fn extend(&mut self, package: &PackageIdent, deps: &[PackageIdent]) -> (usize, usize) {
```
Comment (mpeck) [inline components/common/src/package_graph.rs]: https://rust-lang.github.io/rust-clippy/master/#ptr_arg

File: components/common/src/package_graph.rs [hunk 14]
```diff
@@ -107,10 +107,10 @@ impl PackageGraph {
                 // `skip` it here so it's not in the result Vec
                 let bfs = Bfs::new(&self.graph, idx).iter(&self.graph).skip(1);
 
-                bfs.into_iter()
-                    .map(|child| self.graph.node_weight(child).unwrap())
+                bfs.map(|child| self.graph.node_weight(child).unwrap())
```
Comment (mpeck) [inline components/common/src/package_graph.rs]: https://rust-lang.github.io/rust-clippy/master/#identity_conversion

File: components/common/src/package_graph.rs [hunk 15]
```diff
@@ -107,10 +107,10 @@ impl PackageGraph {
...
                     .collect()
-            }).unwrap_or(vec![])
+            })
+            .unwrap_or_else(Vec::new)
```
Comment (mpeck) [inline components/common/src/package_graph.rs]: https://rust-lang.github.io/rust-clippy/master/#or_fun_call

File: components/common/src/ui.rs [hunk 16]
```diff
@@ -155,7 +155,7 @@ pub trait UIWriter {
         } else {
             format!("{} {}\n", symbol, message)
         };
-        self.out().write(formatted.as_bytes())?;
+        self.out().write_all(formatted.as_bytes())?;
```
Comment (mpeck) [inline components/common/src/ui.rs]: https://rust-lang.github.io/rust-clippy/master/#unused_io_amount

File: components/common/src/ui.rs [hunk 17]
```diff
@@ -256,11 +256,7 @@ pub trait UIWriter {
         T: AsRef<str>,
     {
         if self.is_out_colored() {
-            write!(
-                self.out(),
-                "{}\n",
-                Colour::Green.bold().paint(text.as_ref())
-            )?;
+            writeln!(self.out(), "{}", Colour::Green.bold().paint(text.as_ref()))?;
```
Comment (mpeck) [inline components/common/src/ui.rs]: https://rust-lang.github.io/rust-clippy/master/#write_with_newline

File: components/common/src/ui.rs [hunk 18]
```diff
@@ -432,45 +428,46 @@ impl UIReader for UI {
     }
 
     fn prompt_yes_no(&mut self, question: &str, default: Option<bool>) -> Result<bool> {
-        let ref mut stream = self.shell.out;
+        let stream = &mut self.shell.out;
```
Comment (mpeck) [inline components/common/src/ui.rs]: https://rust-lang.github.io/rust-clippy/master/#toplevel_ref_arg

File: components/common/src/ui.rs [hunk 19]
```diff
@@ -540,14 +537,14 @@ impl UIReader for UI {
     where
         T: fmt::Display,
     {
-        let editor = env::var("EDITOR").map_err(|e| Error::EditorEnv(e))?;
+        let editor = env::var("EDITOR").map_err(Error::EditorEnv)?;
```
Comment (mpeck) [inline components/common/src/ui.rs]: https://rust-lang.github.io/rust-clippy/master/#redundant_closure

File: components/common/src/ui.rs [hunk 20]
```diff
@@ -540,14 +537,14 @@ impl UIReader for UI {
...
 
         let mut tmp_file_path = env::temp_dir();
         tmp_file_path.push(format!("_hab_{}.tmp", Uuid::new_v4()));
 
         let mut tmp_file = File::create(&tmp_file_path)?;
 
-        if contents.len() > 0 {
+        if !contents.is_empty() {
```
Comment (mpeck) [inline components/common/src/ui.rs]: https://rust-lang.github.io/rust-clippy/master/#len_zero

**Comment by mpeck** [comment]
Closing this for now so we can discuss which rules we want to use and which to disregard.

**Pull request closed without merging**